exifinterface/exifinterface/src/main/java/androidx/exifinterface/media/ExifInterface.java

```diff
@@ -6728,8 +6730,11 @@ private void saveWebpAttributes(InputStream inputStream, OutputStream outputStre
 
                         // Retrieve image width/height
                         widthAndHeight = totalInputStream.readInt();
-                        width = (widthAndHeight << 18) >> 18;
-                        height = (widthAndHeight << 2) >> 18;
+                        int width = widthAndHeight & 0x0000FFFF;
+                        int height = (widthAndHeight & 0xFFFF0000) >> 16;
+                        if (width > WEBP_MAX_DIMENSION || height > WEBP_MAX_DIMENSION) {
+                            throw new IOException("Maximum side allowed is 16383 pixels.");
```

Comment:
I think we should inform the caller about the maximum supported dimensions. If we only read 14 bits, the caller won't be able to perceive any abnormality.

Reply:
Not sure I understand. Each dimension is only permitted to be 14 bits, the other 2 bits of the 16 are not dimensions, they are a scaling factor that isn't relevant here, so not reading these 2 bits seems correct to me.

Reply:
Thank you for your review. After carefully studying the RFC 6386 document and the VP8 specification, I realize that my initial understanding was incorrect. You are right; the width and height should indeed only use 14 bits, not 16.

I agree with your suggestion to modify the code to read only 14 bits for width and height. This not only more accurately follows the specification but also automatically limits the maximum dimensions to 16383x16383 pixels without requiring additional checks.

I will update the code according to your recommendation, using 14-bit masks to read the width and height. Additionally, I think it would be beneficial to explicitly state this maximum size limitation in the documentation to ensure callers are aware of this constraint.

Reply:
> Additionally, I think it would be beneficial to explicitly state this maximum size limitation in the documentation to ensure callers are aware of this constraint.

I don't think it's necessary for `ExifInterface`'s documentation to duplicate information that is available in each image format spec (as is the case here). Otherwise it risks getting very unwieldy - since we end up trying to include all the information from the specs of the many image formats we support.

Reply:
I updated my code just now. Please have a look.